mesonbuild/compilers/mixins/gnu.py

```diff
@@ -367,9 +390,9 @@ def has_arguments(self, args, env, code, mode):
                 result = False
             if self.language in {'c', 'objc'} and 'is valid for C++/ObjC++' in p.stderr:
                 result = False
-        return result, p.cached
+            return result, p.cached
```

Comment:
Not sure I understand this change.

Reply:
Ah, I can change that. At one point in the series I was doing something clever to set the cached variable, but then changes my mind and this didn't get changed back